src/Orchard.Web/Modules/Orchard.Autoroute/Services/AutorouteService.cs

```diff
@@ -206,21 +206,26 @@ public class AutorouteService : Component, IAutorouteService {
         }
 
         public bool ProcessPath(AutoroutePart part) {
-            var pathsLikeThis = GetSimilarPaths(part.Path).ToArray();
-
-            // Don't include *this* part in the list
-            // of slugs to consider for conflict detection.
-            pathsLikeThis = pathsLikeThis.Where(p => p.ContentItem.Id != part.ContentItem.Id).ToArray();
-
-            if (pathsLikeThis.Any()) {
-                var originalPath = part.Path;
-                var newPath = GenerateUniqueSlug(part, pathsLikeThis.Select(p => p.Path));
-                part.DisplayAlias = newPath;
-
-                if (originalPath != newPath)
-                    return false;
+            var parthWithSamePath = _contentManager.Query<AutoroutePart, AutoroutePartRecord>()
```

Comment:
How is that better? Can you explain what makes it faster on lots of records?

Also is this query done only on database, or does it load everything and then filters in memory?